### PR TITLE
frontend i18n: Add 'fr' locale for French

### PR DESCRIPTION
**Pull request by illume**
Note to reviewers:
- either leave review comments in the files tab with the suggested change, or leave a comment about a screenshot
- for screenshot comments, please mention the number of the screenshot you're referring to (1-9)

## How to use

Switch to French language in top right 'fr' locale choosing select box.

To run locally with this branch (you'll need golang and nodejs):
```bash
git clone git@github.com:kinvolk/headlamp.git
cd headlamp
git fetch --all
git checkout fr-fr

make backend frontend
make run-backend
```

In a separate terminal:
```bash
make run-frontend
```

Now it should open in the web browser with the clusters configured in your `~/.kube/config`

<br/>1)
Screenshot 2022-12-05 175536

<br/>2)
Screenshot 2022-12-05 175627

<br/>3)
Screenshot 2022-12-05 175652

<br/>4)
Screenshot 2022-12-05 175730

<br/>5)
Screenshot 2022-12-05 175755

<br/>6)
Screenshot 2022-12-05 175835

<br/>7)
Screenshot 2022-12-05 175905

<br/>8)
Screenshot 2022-12-05 175931

<br/>9)
Screenshot 2022-12-05 180020

<br/>10)
Screenshot 2022-12-05 175958

## How to use

Switch to French language in top right 'fr' locale choosing select box.

## Testing done

- [x] Went through different screens.
- [x] Have translations reviewed

### CODE REVIEW

**Comment by joaquimrocha** [comment]
There's a lot of concepts here that shouldn't be translated to French.
E.g. resource conceptual names shouldn't be translated. I know that we have them marked to be translated but this is just in case some of them are and others aren't which may depend on the language.

A good place to check those is the Kubernetes docs. See how [ReplicaSet](https://kubernetes.io/fr/docs/concepts/workloads/controllers/replicaset/) is not translated.

File: frontend/src/i18n/locales/fr/glossary.json [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  "State": "État",
+  "Node": "Nœud",
+  "Host IP": "IP de l'hôte",
+  "Pod IP": "Pod IP",
+  "Restarts": "Redémarre",
+  "Replica Sets": "Sets de répliques",
+  "Generation": "Génération",
+  "Reference Kind": "Type de référence",
+  "Reference Name": "Nom de référence",
+  "Binding Info": "Informations sur la reliure",
```
[the remark this answers is not in the document]
Comment (illume) [reply frontend/src/i18n/locales/fr/glossary.json:87]: It's the title of a section called Binding Info for Role Bindings.
image

**Review by illume — COMMENTED** [review]
Thank you very much @eiffel-fl 

I went through and answered your context questions.

File: frontend/src/i18n/locales/fr/glossary.json [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  "Command": "Commande",
+  "Environment": "Environnement",
+  "Liveness Probes": "Sondes de vivacité",
+  "Volume Mounts": "Supports de volume",
+  "Containers": "Conteneurs",
+  "Container Spec": "Spécifications des conteneurs",
+  "Conditions": "Conditions",
+  "tabs": "onglets",
+  "Container": "Conteneur",
+  "Config Maps": "Cartes de configuration",
```
[the remark this answers is not in the document]
Comment (illume) [reply frontend/src/i18n/locales/fr/glossary.json:27]: Yeah. I'm wondering if this is another one of those Kubernetes terms that shouldn't be translated?

File: frontend/src/i18n/locales/fr/glossary.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (illume) [reply frontend/src/i18n/locales/fr/glossary.json:87]: I see in the kubernetes dashboard they are not translating Cluster Role Bindings
https://github.com/kubernetes/dashboard/blob/1f109d748bdd51726f45766d3713e582e720e890/modules/web/i18n/fr/messages.fr.xlf#L90

(I don't see Role Bindings in there though)

File: frontend/src/i18n/locales/fr/glossary.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (illume) [reply frontend/src/i18n/locales/fr/glossary.json:27]: The kubernetes dashboard doesn't translate it...
https://github.com/kubernetes/dashboard/blob/1f109d748bdd51726f45766d3713e582e720e890/modules/web/i18n/fr/messages.fr.xlf#L835

But does translate "Config and Storage" to Configuration et Stockage
https://github.com/kubernetes/dashboard/blob/1f109d748bdd51726f45766d3713e582e720e890/modules/web/i18n/fr/messages.fr.xlf#L835

**Comment by illume** [comment]
I'll go through and update the Kubernetes terms like they do in the kubernetes docs/translations (and in the es/pt headlamp ones).

**Comment by illume** [comment]
In f88f07e I used the Kubernetes english terms (like Role Binding, Config Maps, and Binding Info).

@eiffel-fl What do you think of them?

**Comment by illume** [comment]
I fixed a few things from your comments here, and also replied:
https://github.com/kinvolk/headlamp/commit/f88f07e17e22bb9584536fe0f746d2b3a7018f62

(I missed these comments somehow because they didn't show up on this PR page... Some github bug I guess.)

**Comment by illume** [comment]
I noticed "Use minimal editor" needs a translation (like in screenshot 5).

In resource.json
```json
  "Use minimal editor": "Utiliser un éditeur minimal"
```

**Comment by illume** [comment]
Thanks heaps @eiffel-fl :)

- applied your patch
- made a small fix for the 'Clear' one. By moving one of them into the notifications.json file. I'll have to fix the other langs (en/es/pt/etc) to use that one later.
- However, I didn't debug the ones which aren't translated yet. Will do that next.

**Comment by illume** [comment]
@eiffel-fl I pushed a couple of fixes for the ones shown in your screen shots. See the bottom of the fr/resource.json

However, I can't find the "Buffer is not defined" message anywhere. I guess it's an error message coming from... somewhere, perhaps in JS. Do you know how to reproduce getting that error?

**Comment by illume** [comment]
Thanks for the patch @eiffel-fl , applied that one too.
Those two messages are translating properly now.

Screenshot 2022-12-23 144907

**Pull request merged**